lib/teleterm/gateway/app_middleware.go

```diff
@@ -49,11 +51,17 @@ func (m *appMiddleware) OnNewConnection(ctx context.Context, lp *alpn.LocalProxy
 
 	m.log.WithError(err).Debug("Gateway certificates have expired")
 
-	return trace.Wrap(m.onExpiredCert(ctx))
+	cert, err := m.onExpiredCert(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	lp.SetCerts([]tls.Certificate{cert})
```

Comment:
nice. I forgot the callback also has a pointer to the local proxy. Thanks for the update.